src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleEqualityTests.cs

```diff
@@ -810,13 +813,138 @@ static void Main()
     {
         (string, string)? t = (null, null);
         System.Console.Write(t == default);
+        System.Console.Write(t != default);
+        System.Console.Write(default == t);
+        System.Console.Write(default != t);
+    }
+}";
+            var comp = CreateCompilation(source, options: TestOptions.DebugExe);
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "FalseTrueFalseTrue");
+
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+
+            var defaultLiterals = tree.GetCompilationUnitRoot().DescendantNodes().OfType<LiteralExpressionSyntax>()
+                .Where(e => e.Kind() == SyntaxKind.DefaultLiteralExpression);
+
+            foreach (var literal in defaultLiterals)
+            {
+                Assert.Equal("default", literal.ToString());
+                var info = model.GetTypeInfo(literal);
+                Assert.Equal("(System.String, System.String)?", info.Type.ToTestDisplayString());
+                Assert.Equal("(System.String, System.String)?", info.ConvertedType.ToTestDisplayString());
+            }
+        }
+
+        [Fact]
+        public void TestNullableTupleAndDefault_Nested()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        (string, string)? t = (null, null);
+        System.Console.Write((null, t) == (null, default));
+        System.Console.Write((t, null) != (default, null));
+    }
+}";
+            var comp = CreateCompilation(source, options: TestOptions.DebugExe);
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "FalseTrue");
+
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+
+            var defaultLiterals = tree.GetCompilationUnitRoot().DescendantNodes().OfType<LiteralExpressionSyntax>()
+                .Where(e => e.Kind() == SyntaxKind.DefaultLiteralExpression);
+
+            foreach (var literal in defaultLiterals)
+            {
+                Assert.Equal("default", literal.ToString());
+                var info = model.GetTypeInfo(literal);
+                Assert.Equal("(System.String, System.String)?", info.Type.ToTestDisplayString());
+                Assert.Equal("(System.String, System.String)?", info.ConvertedType.ToTestDisplayString());
+            }
+        }
+
+        [Fact]
+        public void TestNestedDefaultWithNonTupleType()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        System.Console.Write((null, 1) == (null, default));
```

Comment:
>default [](start = 49, length = 7)

Do we have a test against nullable non-tuple?

Reply:
Added below. 

Note that there is a known bug with `S? s;` used in `s == default`, when `S` has operator `==(S, int)`. This is not specific to tuple equality. Filed [https://github.com/dotnet/roslyn/issues/25318](https://github.com/dotnet/roslyn/issues/25318)


---
In reply to: [173294618](https://github.com/dotnet/roslyn/pull/25327#discussion_r173294618) [](ancestors = 173294618)